drivers/MbedCRC.h

```diff
@@ -0,0 +1,463 @@
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_32BIT_ANSI, 32>::MbedCRC(uint32_t initial_xor, uint32_t final_xor, bool reflect_data, bool reflect_remainder):
+        _initial_value(initial_xor), _final_xor(final_xor), _reflect_data(reflect_data), _reflect_remainder(reflect_remainder),
+        _crc_table((uint32_t *)Table_CRC_32bit_ANSI)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_8BIT_CCITT, 8>::MbedCRC(uint32_t initial_xor, uint32_t final_xor, bool reflect_data, bool reflect_remainder):
+        _initial_value(initial_xor), _final_xor(final_xor), _reflect_data(reflect_data), _reflect_remainder(reflect_remainder),
+        _crc_table((uint32_t *)Table_CRC_8bit_CCITT)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_7BIT_SD, 7>::MbedCRC(uint32_t initial_xor, uint32_t final_xor, bool reflect_data, bool reflect_remainder):
+        _initial_value(initial_xor), _final_xor(final_xor), _reflect_data(reflect_data), _reflect_remainder(reflect_remainder),
+        _crc_table((uint32_t *)Table_CRC_7Bit_SD)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_16BIT_CCITT, 16>::MbedCRC(uint32_t initial_xor, uint32_t final_xor, bool reflect_data, bool reflect_remainder):
+        _initial_value(initial_xor), _final_xor(final_xor), _reflect_data(reflect_data), _reflect_remainder(reflect_remainder),
+        _crc_table((uint32_t *)Table_CRC_16bit_CCITT)
+{
+}
+
+template<>
+MbedCRC<POLY_16BIT_IBM, 16>::MbedCRC(uint32_t initial_xor, uint32_t final_xor, bool reflect_data, bool reflect_remainder):
+        _initial_value(initial_xor), _final_xor(final_xor), _reflect_data(reflect_data), _reflect_remainder(reflect_remainder),
+        _crc_table((uint32_t *)Table_CRC_16bit_IBM)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_32BIT_ANSI, 32>::MbedCRC():
+    _initial_value(~(0x0)), _final_xor(~(0x0)), _reflect_data(true), _reflect_remainder(true),
+    _crc_table((uint32_t *)Table_CRC_32bit_ANSI)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_16BIT_IBM, 16>::MbedCRC():
+    _initial_value(0), _final_xor(0), _reflect_data(true), _reflect_remainder(true),
+    _crc_table((uint32_t *)Table_CRC_16bit_IBM)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_16BIT_CCITT, 16>::MbedCRC():
+    _initial_value(~(0x0)), _final_xor(0), _reflect_data(false), _reflect_remainder(false),
+    _crc_table((uint32_t *)Table_CRC_16bit_CCITT)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_7BIT_SD, 7>::MbedCRC():
+    _initial_value(0), _final_xor(0), _reflect_data(false), _reflect_remainder(false),
+    _crc_table((uint32_t *)Table_CRC_7Bit_SD)
+{
+    mbed_crc_ctor();
+}
+
+template<>
+MbedCRC<POLY_8BIT_CCITT, 8>::MbedCRC():
+    _initial_value(0), _final_xor(0), _reflect_data(false), _reflect_remainder(false),
+    _crc_table((uint32_t *)Table_CRC_8bit_CCITT)
+{
+    mbed_crc_ctor();
+}
+
+#if   defined ( __CC_ARM )
+#elif defined ( __GNUC__ )
+#pragma GCC diagnostic pop
```

Comment:
Can we add a comment about what does it do?

Reply:
I am disabling warnings reported as "Shift count is negative", which is kind of invalid warning.

".\mbed-os\drivers/MbedCRC.h(231): warning:  #62-D: shift count is negative
          detected during instantiation of "std::int32_t mbed::MbedCRC<polynomial, width>::compute_partial_stop(std::uint32_t *) [with polynomial=79764919U, width=(std::uint8_t)' ']"  at line 15 of "main.cpp""

Code
```        if ((width < 8) && (NULL == _crc_table)) {
            p_crc = (uint32_t)(p_crc << (8 - width));
        }
```
Compiler warns of the shift operation with `width` as it is width=(std::uint8_t), but we check for ( width < 8) before performing shift, so it should not be an issue.

Do we add such detailed information in code?

Reply:
If you just add one sentence description, it will save combined hours of googling over couple of years ;)

Reply:
Oh that's switching it off... ignore me